src/components/BottomBarWithIcons/index.js

```diff
@@ -23,193 +22,156 @@ import {
   selectAddIcon,
   selectRemoveIcon,
   resetEditIcon,
-} from 'features/toggleBottomBarIconsSlice/toggleBottomBarIconsSlice'
+} from 'features/toggleBottomBarIcons/toggleBottomBarIconsSlice'
 import { selectBasketHistory } from 'features/createBasketHistory/createBasketHistorySlice'
+import { selectGroceriesBasketData } from 'features/groceries/groceriesSlice'
+import { selectProductsBasketData } from 'features/products/productsSlice'
+import { selectDishesBasketData } from 'features/dishes/dishesSlice'
+import { selectSavedListBasketData } from 'features/savedList/savedListSlice'
+import { postHistoryBasket } from 'features/history/historySlice'
 
 function BottomBarWithIcons({ icons, saveIconArray }) {
-  const useBasketGroceres = API.useBasketGroceres()
-  const useBasketProducts = API.useBasketProducts()
-  const useBasketDishes = API.useBasketDishes()
-  const useBasketSavedLists = API.useBasketSavedLists()
+  const dispatch = useDispatch()
+  const history = useHistory()
 
-  const [mutate_Post_History] = API.useAddHistory()
+  const groceriesBasketData = useSelector(selectGroceriesBasketData)
```

Comment:
po prostu groceries, products, dishes, savedList. Unikaj długich nazw, kiedy jedno słowo już powie wszystko co trzeba

Reply:
grocenies wygląda ma kilka różnych danych (data,categoriesData,basketData)

a co zamiast robić selektory do wszystkich dany to zrobił bym taki ?
export const groceries = (state) => state.groceries
  const groceries = useSelector(selectGroceries)

console.log(groceries.data)
  console.log(groceries.categoriesData)
  console.log(groceries.basketData)
  console.log(groceries.dataStatus)
  console.log(groceries.basketDataStatus)

widać w konsoli wiec działa, Zamiast tworzyc kazdy selektor do danych to bym miał jeden :) co myślisz ?